src/smsfusion/benchmark/_benchmark.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+class BeatSignal:
+    """
+    Generate an evenly sampled beating sinusoidal signal by defining a main
+    frequency and a beating frequency.
+
+    Parameters
+    ----------
+    f_main : float
+        The main sinusoidal signal frequency.
+    f_beat : float
+        The beating signal frequency.
+    freq_hz : bool, default True.
+        If ``True``, ``f_main`` and ``f_beat`` are in Hz. Otherwise,
+        rad/s is assumed.
+
+    """
+
+    def __init__(self, f_main: float, f_beat: float, freq_hz: bool = True) -> None:
+        self._f_main = f_main
+        self._f_beat = f_beat
+
+        if freq_hz:
+            self._f_main *= 2.0 * np.pi
+            self._f_beat *= 2.0 * np.pi
+
+    def __call__(
+        self,
+        fs: float,
+        n: int,
+        amp: float = 5.0,
```

Comment:
Why 5? Would it make more sense to have a default amplitude of 1?

Reply:
Also, it this parameter necessary? Isn't it easier to just generate a signal which has an amplitude of 1, and then it is up to the user to scale the signal if needed?